unlock-app/src/components/interface/checkout/main/Payment.tsx

```diff
@@ -91,24 +119,25 @@ export function Payment({ injectedProvider, checkoutService }: Props) {
     }
   )
 
-  const networkConfig = config.networks[lock.network]
-
-  const isWaiting = isLoading || isCrossmintLoading || isBalanceLoading
+  const isWaiting =
+    isLoading || isCrossmintLoading || isBalanceLoading || isPricingDataLoading
 
   const isReceiverAccountOnly =
     recipients.length <= 1 &&
     recipients[0]?.toLowerCase() === account?.toLowerCase()
 
-  const enableCrypto = !isUnlockAccount || !!balance?.isPayable
+  const enableCrypto = !isUnlockAccount
 
   const enableClaim: boolean = !!(
     canClaim &&
     !isCanClaimLoading &&
     isReceiverAccountOnly &&
-    !balance?.isPayable
+    !balance?.isGasPayable
   )
 
-  const { data: routes, isInitialLoading: isUniswapRoutesLoading } =
+  const canAfford = balance?.isGasPayable && balance?.isPayable
```

Comment:
will that check if account has enough to pay for gas+price. It seems it checks for it separately

Reply:
Indeed, we need to be better eventually to check for the actual gas spent. 
- For non native tokens it checks that the balance is enough + there is _some_ native to pay for gas.
- For native tokens it checks there is indeed more than the cost of the membership, but not that there is enough to _also_ pay for gas if there is just enough to pay the membership.
The wallet will show error messages in both cases though.